# Name: Veleña, Harold V.

<hr>Analyst: Harold V. Veleña</hr>

In [1]:
pip install plotly

Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip is available: 24.3.1 -> 25.2
[notice] To update, run: python.exe -m pip install --upgrade pip


In [31]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

<p>Importing dataset to Jupyter notebook</p>
<p></p>

In [32]:
a = np.array([2.2, 3, 4])
a.dtype


dtype('float64')

# Getting the penguins data

In [33]:
penguins = pd.read_csv('penguins.csv')

penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

# How to create a Column

In [34]:
penguins["mass_kg"] = round(penguins["body_mass_g"] / 1000,1)
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  mass_kg  
0         3750.0    male  2007      3.8  
1         3800.0  female  2007      3.8  
2         3250.0  female  2007      3.2  
3            NaN     NaN  2007      NaN  
4         3450.0  female  2007      3.4  
..           ...     ...   ...      ...  
339       4000.0    male  2009      4.0  
340       3400.0  female  2009      3.4  
341       3775.0    male  2009      3.8  
342       4100.0    male  2009      4.1  
343       3775.0  female  2009      3.8  

[344 rows x 9 columns]

# Descriptive Analysis

<h2>Getting the Top 5</h2>

In [35]:
penguins.head()

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0  Adelie  Torgersen            39.1           18.7              181.0   
1  Adelie  Torgersen            39.5           17.4              186.0   
2  Adelie  Torgersen            40.3           18.0              195.0   
3  Adelie  Torgersen             NaN            NaN                NaN   
4  Adelie  Torgersen            36.7           19.3              193.0   

   body_mass_g     sex  year  mass_kg  
0       3750.0    male  2007      3.8  
1       3800.0  female  2007      3.8  
2       3250.0  female  2007      3.2  
3          NaN     NaN  2007      NaN  
4       3450.0  female  2007      3.4

In [36]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            344 non-null    object 
 1   island             344 non-null    object 
 2   bill_length_mm     342 non-null    float64
 3   bill_depth_mm      342 non-null    float64
 4   flipper_length_mm  342 non-null    float64
 5   body_mass_g        342 non-null    float64
 6   sex                333 non-null    object 
 7   year               344 non-null    int64  
 8   mass_kg            342 non-null    float64
dtypes: float64(5), int64(1), object(3)
memory usage: 24.3+ KB


In [37]:
penguins.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      342.000000     342.000000         342.000000   342.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.459584       1.974793          14.061714   801.954536   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.225000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year     mass_kg  
count   344.000000  342.000000  
mean   2008.029070    4.202047  
std       0.818356    0.803089  
min    2007.000000    2.700000  
25%    2007.000000    3.600000  
50%    2008.000000    4.000000  
75%    2009.000000    4.800000  
max    2009.000000    6.300000

In [38]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
mass_kg               2
dtype: int64

In [39]:
penguins[penguins["island"] == 'Dream']

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
30      Adelie  Dream            39.5           16.7              178.0   
31      Adelie  Dream            37.2           18.1              178.0   
32      Adelie  Dream            39.5           17.8              188.0   
33      Adelie  Dream            40.9           18.9              184.0   
34      Adelie  Dream            36.4           17.0              195.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  mass_kg  
30        3250.0  female  2007      3.2  
31        3900.0    male  2007      3.9  
32        3300.0  female  2007      3.3  
33        3900.0    male  2007      3.9  
34        3325.0  female  2007      3.3  
..           ...     ...   ...      ...  
339       4000.0    male  2009      4.0  
340       3400.0  female  2009      3.4  
341       3775.0    male  2009      3.8  
342       4100.0    male  2009      4.1  
343       3775.0  female  2009      3.8  

[124 rows x 9 columns]

In [40]:
penguins["bill_length_mm"].isnull()

0      False
1      False
2      False
3       True
4      False
       ...  
339    False
340    False
341    False
342    False
343    False
Name: bill_length_mm, Length: 344, dtype: bool

In [41]:
penguins["species"].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

In [42]:
Adelie_pd = penguins[penguins["species"] == 'Adelie']
Gentoo_pd = penguins[penguins["species"] == 'Gentoo']
Crinstrap_pd = penguins[penguins["species"] == 'Chinstrap']

In [43]:
Adelie_pd

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
147  Adelie      Dream            36.6           18.4              184.0   
148  Adelie      Dream            36.0           17.8              195.0   
149  Adelie      Dream            37.8           18.1              193.0   
150  Adelie      Dream            36.0           17.1              187.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g     sex  year  mass_kg  
0         3750.0    male  2007      3.8  
1         3800.0  female  2007      3.8  
2         3250.0  female  2007      3.2  
3            NaN     NaN  2007      NaN  
4         3450.0  female  2007      3.4  
..           ...     ...   ...      ...  
147       3475.0  female  2009      3.5  
148       3450.0  female  2009      3.4  
149       3750.0    male  2009      3.8  
150       3700.0  female  2009      3.7  
151       4000.0    male  2009      4.0  

[152 rows x 9 columns]

In [44]:
Gentoo_pd

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152  Gentoo  Biscoe            46.1           13.2              211.0   
153  Gentoo  Biscoe            50.0           16.3              230.0   
154  Gentoo  Biscoe            48.7           14.1              210.0   
155  Gentoo  Biscoe            50.0           15.2              218.0   
156  Gentoo  Biscoe            47.6           14.5              215.0   
..      ...     ...             ...            ...                ...   
271  Gentoo  Biscoe             NaN            NaN                NaN   
272  Gentoo  Biscoe            46.8           14.3              215.0   
273  Gentoo  Biscoe            50.4           15.7              222.0   
274  Gentoo  Biscoe            45.2           14.8              212.0   
275  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  mass_kg  
152       4500.0  female  2007      4.5  
153       5700.0    male  2007      5.7  
154       4450.0  female  2007      4.4  
155       5700.0    male  2007      5.7  
156       5400.0    male  2007      5.4  
..           ...     ...   ...      ...  
271          NaN     NaN  2009      NaN  
272       4850.0  female  2009      4.8  
273       5750.0    male  2009      5.8  
274       5200.0  female  2009      5.2  
275       5400.0    male  2009      5.4  

[124 rows x 9 columns]

In [45]:
Crinstrap_pd

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  mass_kg  
276       3500.0  female  2007      3.5  
277       3900.0    male  2007      3.9  
278       3650.0    male  2007      3.6  
279       3525.0  female  2007      3.5  
280       3725.0    male  2007      3.7  
..           ...     ...   ...      ...  
339       4000.0    male  2009      4.0  
340       3400.0  female  2009      3.4  
341       3775.0    male  2009      3.8  
342       4100.0    male  2009      4.1  
343       3775.0  female  2009      3.8  

[68 rows x 9 columns]

In [46]:
penguins[(penguins["species"] == 'Chinstrap') & (Crinstrap_pd["body_mass_g"] >= 4000)]

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
284  Chinstrap  Dream            46.0           18.9              195.0   
289  Chinstrap  Dream            52.0           18.1              201.0   
291  Chinstrap  Dream            50.5           19.6              201.0   
295  Chinstrap  Dream            49.2           18.2              195.0   
301  Chinstrap  Dream            52.0           19.0              197.0   
305  Chinstrap  Dream            52.8           20.0              205.0   
307  Chinstrap  Dream            54.2           20.8              201.0   
309  Chinstrap  Dream            51.0           18.8              203.0   
313  Chinstrap  Dream            52.0           20.7              210.0   
315  Chinstrap  Dream            53.5           19.9              205.0   
321  Chinstrap  Dream            50.8           18.5              201.0   
323  Chinstrap  Dream            49.0           19.6              212.0   
329  Chinstrap  Dream            50.7           19.7              203.0   
333  Chinstrap  Dream            49.3           19.9              203.0   
339  Chinstrap  Dream            55.8           19.8              207.0   
342  Chinstrap  Dream            50.8           19.0              210.0   

     body_mass_g     sex  year  mass_kg  
284       4150.0  female  2007      4.2  
289       4050.0    male  2007      4.0  
291       4050.0    male  2007      4.0  
295       4400.0    male  2007      4.4  
301       4150.0    male  2007      4.2  
305       4550.0    male  2008      4.6  
307       4300.0    male  2008      4.3  
309       4100.0    male  2008      4.1  
313       4800.0    male  2008      4.8  
315       4500.0    male  2008      4.5  
321       4450.0    male  2009      4.4  
323       4300.0    male  2009      4.3  
329       4050.0    male  2009      4.0  
333       4050.0    male  2009      4.0  
339       4000.0    male  2009      4.0  
342       4100.0    male  2009      4.1

In [47]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
mass_kg               2
dtype: int64

# How to remove null values

In [48]:
penguins["bill_length_mm"].mean()

np.float64(43.9219298245614)

In [49]:
penguins["bill_length_mm"] = penguins["bill_length_mm"].fillna(penguins["bill_length_mm"].mean())

In [50]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
mass_kg               2
dtype: int64

# If the data is categorical we can use mode()

In [51]:
penguins["sex"] = penguins["sex"].fillna(penguins["sex"].mode()[0])

In [52]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        2
flipper_length_mm    2
body_mass_g          2
sex                  0
year                 0
mass_kg              2
dtype: int64

# Repeat the process of removing the null values to the other columns

In [64]:
penguins["bill_depth_mm"] = penguins["bill_depth_mm"].fillna(penguins["bill_depth_mm"].mean())
penguins["flipper_length_mm"] = penguins["flipper_length_mm"].fillna(penguins["flipper_length_mm"].mean())
penguins["body_mass_g"] = penguins["body_mass_g"].fillna(penguins["body_mass_g"].mean())
penguins["mass_kg"] = penguins["mass_kg"].fillna(penguins["mass_kg"].mean())

# All the null values have been removed

In [65]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
mass_kg              0
dtype: int64

# Changing the data type of the columns

In [55]:
penguins["species"] = penguins["species"].astype("category")
penguins["year"] = penguins["year"].astype("category")
penguins["island"] = penguins["island"].astype("category")
penguins["sex"] = penguins["sex"].astype("category")

In [56]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    category
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    category
 7   year               344 non-null    category
 8   mass_kg            342 non-null    float64 
dtypes: category(4), float64(5)
memory usage: 15.4 KB


In [57]:
penguins.columns

Index(['species', 'island', 'bill_length_mm', 'bill_depth_mm',
       'flipper_length_mm', 'body_mass_g', 'sex', 'year', 'mass_kg'],
      dtype='object')

# Exploration Data Analytics (EDA)

<h2>1. Which species has the heviest body mass in kg</h2>

In [58]:
Crinstrap_pd["body_mass_g"].mean()

np.float64(3733.0882352941176)

In [59]:
Adelie_pd["body_mass_g"].mean()

np.float64(3700.662251655629)

In [60]:
Gentoo_pd["body_mass_g"].mean()

np.float64(5076.016260162602)

<h1>Insight #1</h1>
<h1>The heaviest species from the dataset is the GENTOO species</h1>

# 7 Questions activity

<hr>

# 1 What species has the highest population?

In [142]:
penguins["species"].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

<h2>Results:</h2>
<li>Adelie has the highest population in compared to the rest</li>
<li>Chinstrap has the lowest population</li>
<hr>

# 2 What species occurs more often in every island?

<h2>The Adelia species in each island</h2>

In [162]:
Adelie = penguins[penguins["species"] == "Adelie"]
Adelie["island"].value_counts()

island
Dream        56
Torgersen    52
Biscoe       44
Name: count, dtype: int64

<h2>The Chinstrap species in each island</h2>

In [163]:
Chinstrap = penguins[penguins["species"] == "Chinstrap"]
Chinstrap["island"].value_counts()

island
Dream        68
Biscoe        0
Torgersen     0
Name: count, dtype: int64

<h2>The Gentoo species in each island</h2>

In [161]:
Gentoo = penguins[penguins["species"] == "Gentoo"]
Gentoo["island"].value_counts()

island
Biscoe       124
Dream          0
Torgersen      0
Name: count, dtype: int64

<h2>Results:</h2>
<li>Adelie has the highest population in Dream island</li>
<li>Chinstrap has the highest population in Dream island since it is the only species there</li>
<li>Gentoo has the highest population in Biscoe island since it is the only species there</li>
<hr>

# 3 What island has the most Crinstrap?

In [150]:
penguins["island"].value_counts().sum()

np.int64(344)

<h2>Chinstrap in Dream island</h2>

In [121]:
Chinstrap_1 = penguins[penguins["species"] == "Chinstrap"]
common_Chinstrap_1 = Chinstrap_1[Chinstrap_1["island"] == "Dream"]  
common_Chinstrap_1["species"].value_counts()

species
Chinstrap    68
Adelie        0
Gentoo        0
Name: count, dtype: int64

<h2>Chinstrap in Torgersen island</h2>

In [126]:
Chinstrap_2 = penguins[penguins["species"] == "Chinstrap"]
common_Chinstrap_2 = Chinstrap_2[Chinstrap_2["island"] == "Torgersen"]  
common_Chinstrap_2["species"].value_counts()

species
Adelie       0
Chinstrap    0
Gentoo       0
Name: count, dtype: int64

<h2>Chinstrap in Biscoe island</h2>

In [118]:
Chinstrap_3 = penguins[penguins["species"] == "Chinstrap"]
common_Chinstrap_3 = Chinstrap_3[Chinstrap_3["island"] == "Biscoe"]  
common_Chinstrap_3["species"].value_counts()

species
Adelie       0
Chinstrap    0
Gentoo       0
Name: count, dtype: int64

<h3>Results:</h3>
<li>We found out that there are no Chinstrap in Torgersen and Biscoe island</li>
<li>We found out that Chinstrap is abundant in Dream island</li>
<hr>

# 4 What is the average size of flippers for all the species?

In [201]:
Chinstrap = penguins[penguins["species"] == "Chinstrap"]["flipper_length_mm"].mean()
Adelie = penguins[penguins["species"] == "Adelie"]["flipper_length_mm"].mean()
Gentoo = penguins[penguins["species"] == "Gentoo"]["flipper_length_mm"].mean()

print("The average flipper size of the Chinstrap species", round(Chinstrap,1))
print("The average flipper size of the Adelie species", round(Adelie,1))
print("The average flipper size of the Gentoo species", round(Gentoo,1))

The average flipper size of the Chinstrap species 195.8
The average flipper size of the Adelie species 190.0
The average flipper size of the Gentoo species 217.1


<h2>Results:</h2>
<li>The average size of flippers for each species range from 190 to 217 in lenght</li>
<li>Gentoo has the largest average size compare to the other species</li>
<li>Adelie has the smallest average size compared to the other species</li>
<hr>

# 5 What island has the most occuring penguin with the weight over 4.5 Kg above?

<h2>The weight of penguins above 4.5 Kg in Dream island</h2>

In [213]:
dream_mass = penguins[(penguins["island"] == "Dream") & (penguins["mass_kg"] >= 4.5)]
dream_mass["island"].value_counts()

island
Dream        6
Biscoe       0
Torgersen    0
Name: count, dtype: int64

<h2>The weight of penguins above 4.5 Kg in Torgersen island</h2>

In [220]:
Torgersen_mass = penguins[(penguins["island"] == "Torgersen") & (penguins["mass_kg"] >= 4.5)]
Torgersen_mass["island"].value_counts()

island
Torgersen    3
Biscoe       0
Dream        0
Name: count, dtype: int64

<h2>The weight of penguins above 4.5 Kg in Biscoe island</h2>

In [218]:
Biscoe_mass = penguins[(penguins["island"] == "Biscoe") & (penguins["mass_kg"] >= 4.5)]
Biscoe_mass["island"].value_counts()

island
Biscoe       110
Dream          0
Torgersen      0
Name: count, dtype: int64

<h2>Results:</h2>
<li>Biscoe island has the most amount of penguins above 4.5 Kg</li>
<li>Torgersen has the lowest among the other islands</li>
<hr>

# 6 how many penguins are there in dream island?

In [141]:
dream_penguins = penguins[penguins["island"] == "Dream"]
species = dream_penguins["species"].value_counts()
total = species.sum()

print(species)
print("Total penguins in Dream island", total)

species
Chinstrap    68
Adelie       56
Gentoo        0
Name: count, dtype: int64
Total penguins in Dream island 124


<h2>Results:</h2>
<li>There are no Gentoo in Dream island</li>
<li>There are a total of 124 penguins in Dream island</li>
<hr>

# 7 What is the most common sex on each island??

<h2>Number of males and females on Dream Island</h2>

In [171]:
dream_penguins = penguins[penguins["island"] == "Dream"]
dream_penguins["sex"].value_counts()

sex
male      63
female    61
Name: count, dtype: int64

<h2>Number of males and females on Torgersen Island</h2>

In [172]:
dream_penguins = penguins[penguins["island"] == "Torgersen"]
dream_penguins["sex"].value_counts()

sex
male      28
female    24
Name: count, dtype: int64

<h2>Number of males and females on Biscoe Island</h2>

In [174]:
dream_penguins = penguins[penguins["island"] == "Biscoe"]
dream_penguins["sex"].value_counts()

sex
male      88
female    80
Name: count, dtype: int64

<h2>Results:</h2>
<li>There are more male penguins in each islands</li>
<li>Biscoe island has the highest male count compare to the onther islands</li>
<li>Torgersen island has the lowest male count among the other islands</li>
<hr>